app/presenters/search_result_presenter.rb

```diff
@@ -8,29 +8,80 @@ class SearchResultPresenter
            :es_score,
            to: :search_result
 
-  def initialize(search_result, metadata)
+  def initialize(search_result, metadata, doc_index, doc_count, finder_name, debug_score, highlight)
```

Comment:
This is a few arguments, is it possible we could introduce a new subclass for the metadata? If we already have one, could the additional logic related to `structure_metadata` be handled by the metadata presenter?

If we need to give this many methods, could we add a single hash argument so we don't need to pass primitive types? E.g. `def initialize(result:, metadata, doc_index:, doc_count:, finder_name:, debug_score:, highlight: false)`

Reply:
The little I understand of this sounds good! Let's catch up offline so you can give me some pointers.

Reply:
I've changed `search_result_presenter` to only have a single hash argument as suggested.

Regarding the metadata, I had a crack at it (it does seem unnecessary to call `metadata = metadata_presenter_class.new(result.metadata).present` right before passing it somewhere else) but I don't know where `metadata_presenter_class` comes from - any idea how I can define that in `search_result_presenter`?

Reply:
Having looked into this a bit deeper, I agree that this needs to be addressed but I'd like to suggest we do so in a separate PR - this change feels potentially big and this is a big PR already.